src/crewai/memory/storage/rag_storage.py

```diff
@@ -100,31 +80,46 @@ def search(  # type: ignore # BUG?: Signature of "search" incompatible with supe
     ) -> List[Any]:
         if not hasattr(self, "app"):
             self._initialize_app()
-        from embedchain.vectordb.chroma import InvalidDimensionException
 
         with suppress_logging():
-            try:
-                results = (
-                    self.app.search(query, limit, where=filter)
-                    if filter
-                    else self.app.search(query, limit)
-                )
-            except InvalidDimensionException:
-                self.app.reset()
-                return []
-        return [r for r in results if r["metadata"]["score"] >= score_threshold]
+            response = self.collection.query(query_texts=query, n_results=limit)
```

Comment:
_Suggestion_: Maybe still makes sense to add a try/except here to catch any possible error being raised